### ISSUE TITLE
Running multiple lines of python via tohil::exec/eval is a bit clunky

### ISSUE THREAD

**Issue by NasaGeek** [flightaware/tohil#48]
```tcl
% package require tohil
4.1.0
% tohil::exec {
    def fn():
        pass
}
unexpected indent (<string>, line 2)
%
```
Python wants your globally scoped code to have no extraneous indentation.

This issue doesn't demonstrate terribly well in a shell where it's silly to indent your code, but consider the case where you're running some python within a tcl proc. Suddenly you'll need to remove all indentation from your python code for it to run correctly and you're left with this unfortunate-looking code
```tcl
proc dostuff {} {
	tohil::exec {
def fn():
	pass
}
}
```

tcllib's [undent command](https://core.tcl-lang.org/tcllib/doc/trunk/embedded/md/tcllib/files/modules/textutil/textutil.md#6) can handle removing the leading indentation properly, as can [python's dedent function](https://docs.python.org/3/library/textwrap.html#textwrap.dedent). Is there any way this whitespace can be stripped automatically in tohil's exec and eval commands? I'd accept an opt-in flag for the behavior or even a utility proc (I'm just not terribly interested in forcing a tcllib dependency for this).

**Comment by lehenbauer** [comment]
Hmm, yeah, pretty easy to solve in C to avoid a dependency.  I agree this seems like a good idea.  I'm wondering since it's an IndentationError to have any extraneous indent in there, if it even needs a switch or if it should just function that way all the time.

**Comment by lehenbauer** [comment]
How about if we make the matching more stringent than undent, insofar as whatever whitespace we see on the first line is expected to be present on each additional line (except blank lines, which we'll ignore) and if it isn't, we'll error?

**Comment by lehenbauer** [comment]
The more stringent approach fared well in private conversations with nasageek and another.  I foresee something in C that figures out the whitespace of the first line that isn't empty, copies it sans leading whitspace and all of any following lines stripping the same whitespace, making it an error if any of the following non-empty lines don't have the exact same whitespace, making a new string or Tcl object or dstring or whatever, containing the unindented code.

If it immediately detects that there is no leading whitespace on the first non-empty line, it might ought to give up right away and avoid copying the string if that's easy to implement.

I think this should only apply to tohil::exec as I don't see a legal way to call tohil::eval with a multiline sequence unless there's like a triple-quoted string argument or something, but even so, we about for sure don't want to do any unindenting on such a string.

**Comment by resuna** [comment]
What I've done in the past is something like:

> proc dostuff {} {
> 	tohil::exec {
>                 |def fn():
>                 |	pass
>         }
> }

Where if the first nonblank character is | it deletes `$whitespace|` from the beginning of every line.

**Comment by NasaGeek** [comment]
I'd rather have the implicit enforcement of consistent leading whitespace without the need for extra marker characters.

**Comment by resuna** [comment]
This was for generated HTML and I found the markers really helpful in keeping track of the indentation when working on the code.

**Comment by lehenbauer** [comment]
Fixed by PR #61
